radixdlt/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -17,50 +17,100 @@
 
 package com.radixdlt.consensus;
 
+import com.google.common.collect.Maps;
 import com.google.inject.Inject;
 import com.radixdlt.consensus.validators.ValidationState;
 import com.radixdlt.consensus.validators.ValidatorSet;
 import com.radixdlt.crypto.ECDSASignature;
+import com.radixdlt.crypto.ECPublicKey;
 import com.radixdlt.crypto.Hash;
+import com.radixdlt.utils.Pair;
 
 import java.util.HashMap;
 import java.util.Objects;
 import java.util.Optional;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 /**
  * Manages pending votes for various vertices
  */
 public final class PendingVotes {
-	private final HashMap<Hash, ValidationState> votes = new HashMap<>();
+	private static final Logger log = LogManager.getLogger();
+
+	private final HashMap<View, Pair<Hash, ValidationState>> state = Maps.newHashMap();
 	private final Hasher hasher;
 
 	@Inject
 	public PendingVotes(Hasher hasher) {
 		this.hasher = Objects.requireNonNull(hasher);
 	}
 
+	/**
+	 * Prepare for a round of voting on the specified vote data by the specified validator set.
+	 *
+	 * @param voteData The data to be voted on
+	 * @param validatorSet The {@link ValidatorSet} that will be voting
+	 */
+	public void startVotingOn(VoteData voteData, ValidatorSet validatorSet) {
+		View voteView = voteData.getProposed().getView();
+		final Hash voteHash = hasher.hash(voteData);
+		this.state.put(voteView, Pair.of(voteHash,  validatorSet.newValidationState(voteHash)));
+	}
+
+	/**
+	 * Finish a round of voting for the specified view.
+	 *
+	 * @param view The view to finish voting for
+	 */
+	public void finishVotingFor(View view) {
+		this.state.remove(view);

```

Comment:
Is it an error here if there is no `view` in `state`? It seems like that shouldn't happen.

Reply:
No, no error as things currently stand, as the state is removed once a quorum is formed at the moment, but the timeout triggering this will still happen regardless.